otx/algorithms/common/utils/data_utils.py

```diff
@@ -0,0 +1,55 @@
+"""Collections of Dataset utils for common OTX algorithms."""
+
+# Copyright (C) 2022 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+from mpa.utils.logger import get_logger
+
+logger = get_logger()
+
+
+def get_cls_img_indices(labels, dataset):
+    """Function for getting image indices per class.
+
+    Args:
+        labels (List[LabelEntity]): List of labels
+        dataset(DatasetEntity): dataset entity
+    """
+    img_indices = {label.name: [] for label in labels}
+    for i, item in enumerate(dataset):
+        item_labels = item.annotation_scene.get_labels()
+        for i_l in item_labels:
+            if i_l in labels:
+                img_indices[i_l.name].append(i)
+
+    return img_indices
+
+
+def get_old_new_img_indices(labels, new_classes, dataset):
```

Comment:
The reason why this method is located in `common/utils/data_utils.py` is our default learning method is `class_incremental`? I'm not sure that methods related to `class_incremental` are exposed on `common`.

Even if we selected `class_incremental` as a base, it is unfamiliar to developers who used a conventional framework.

Reply:
This is something I agree with to some extent. 
This needs to be cleaned up, but with the current structure, it is most efficient to proceed in the training task. 

I think this will also be included in the backlog of future work. Thanks for the good comments.

Reply:
I agree, let's remain some refactorings related to architecture.